src/main/java/crazypants/enderio/machine/farm/TileFarmStation.java

```diff
@@ -643,13 +655,23 @@ public void readCommon(NBTTagCompound nbtRoot) {
     }
     int slotLayoutVersion = nbtRoot.getInteger("slotLayoutVersion");
     if (slotLayoutVersion < 1) {
-      for (int i = inventory.length - 2; i >= 2; i--) {
+      // update from v0 (2+4+4+1) to v1 (3+4+4+1) by moving in+out+cap one back
+      for (int i = (3+4+4+1) - 2; i >= 2; i--) {
         if (inventory[i+1] == null) {
           // Should always be true, but better safe than deleting items
           inventory[i+1] = inventory[i];
           inventory[i] = null;
         }
       }
+      slotLayoutVersion = 1;
+    }
+    if (slotLayoutVersion < 2) {
+      // update from v1 (3+4+4+1) to v2 (3+4+6+1) by moving cap to new position
+      if (inventory[(3+4+6+1)-1] == null) {
+        inventory[(3+4+6+1)-1] = inventory[(3+4+4+1)-1];
+        inventory[(3+4+4+1)-3] = null;
```

Comment:
because that only gives the current implementation. once the layout updates again, it won't give good data. however, that code already is outdated, see latest PR